### ISSUE TITLE
Severe Memory Leak, Possibly Related to UIA

### ISSUE THREAD

**Issue by Simon818** [microsoft/terminal#16209]
### Windows Terminal version

1.19.2682.0

### Windows build number

10.0.19045.3570

### Other Software

NVDA 2023.3

### Steps to reproduce

With NVDA running, open several terminal windows and leave them running for a few hours. Occasionally check the usage of the WindowsTerminal.exe process. The terminal windows do not need to be particularly active, though they should have something more than just a command prompt window running on them--perhaps SSH sessions or WSL windows.

### Expected Behavior

Windows Terminal should remain within reasonable memory usage.

### Actual Behavior

On my system, Windows Terminal's memory usage grows to hundreds of megabytes over time, causing responsiveness issues in Terminal and seemingly affecting the responsiveness of NVDA's UIA access. If I leave a terminal window open overnight, it and all other Terminal windows are nearly unusable the next morning, and the responsiveness of NVDA itself is affected, leaving core OS functionality unusable until I close the process. This leads me to believe the issue is related to UIA.

CC @carlos-zamora @codeofdusk

**Comment by lhecker** [comment]
Possible cause: This doesn't release any memory
https://github.com/microsoft/terminal/blob/main/src/renderer/uia/UiaRenderer.cpp#L234

Limit to either 4KiB or 2x the previous write size, whatever is larger:
```cpp
if (_newOutput.capacity() >= std::max<size_t>(4096, _queuedOutput.size() * 2))
{
    _newOutput = std::wstring{};
    _newOutput.reserve(2048);
}
```

**Comment by carlos-zamora** [comment]
Thanks @Simon818. Could you capture a memory dump for us? We have a guide for that right here: https://github.com/microsoft/terminal/wiki/Troubleshooting-Tips#capturing-and-sending-dumps

**Comment by zadjii-msft** [comment]
I'm gonna give @Simon818 another week to try and get back to us here. This sounds really painful, but we need to get a dump to see where these allocations are coming from.

I suppose @carlos-zamora you might also be able to use App Verifier pointed at the Terminal with NVDA turned on. That might catch some of the more obvious mistakes 🤷

**Comment by zadjii-msft** [comment]
I'm gonna close this in favor of #16217, since that thread has more of an investigation in it. Thanks!

/dup #16217
